config/routes.rb

```diff
@@ -30,12 +30,11 @@
   get "*slug", slug: %r{help/.+}, to: "help#show", constraints: FormatRoutingConstraint.new('help_page')
 
   # Done pages
-  get "*slug", slug: %r{done/.+}, to: "root#publication"
-
-  # Transaction finished pages
-  constraints(slug: /(transaction-finished|driving-transaction-finished)/) do
-    get "/:slug.json"      => redirect("/api/%{slug}.json")
-    get "/:slug(.:format)" => "root#legacy_completed_transaction"
+  constraints FormatRoutingConstraint.new('completed_transaction') do
+    with_options(to: "completed_transaction#show") do
+      get "/:slug(.:format)", slug: /(transaction-finished|driving-transaction-finished)/ # Legacy
```

Comment:
Could this not have it's own controller method? Seems that it's entirely different to the common case.